src/bin/purge_db.rs

```diff
@@ -0,0 +1,111 @@
+use rocksdb::IteratorMode;
+use ssz::Decode;
+use structopt::StructOpt;
+
+use ethportal_api::types::accumulator::EpochAccumulator;
+use ethportal_api::types::block_body::BlockBody;
+use ethportal_api::types::block_header::BlockHeaderWithProof;
+use ethportal_api::types::content_key::HistoryContentKey;
+use ethportal_api::types::receipts::BlockReceipts;
+use trin_core::portalnet::storage::{PortalStorage, PortalStorageConfig};
+use trin_core::portalnet::types::messages::ProtocolId;
+use trin_core::utils::db::get_data_dir;
+
+pub fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let purge_config = PurgeConfig::from_args();
+
+    let mut encoded = hex::decode(&purge_config.private_key).unwrap();
+    let enr_key = CombinedKey::secp256k1_from_bytes(&mut encoded).unwrap();
+    let enr = EnrBuilder::new("v4").build(&enr_key).unwrap();
+    let node_id = enr.node_id();
+    let data_dir_path = get_data_dir(node_id);
+    println!("Purging data for NodeID: {}", node_id);
+    println!("DB Path: {:?}", data_dir_path);
+
+    // Capacity is 0 since it (eg. for data radius calculation) is irrelevant when only removing data.
+    let capacity = 0;
+    let protocol = ProtocolId::History;
+    let config = PortalStorageConfig::new(capacity, node_id, false);
+    let storage = PortalStorage::new(config.clone(), protocol).unwrap();
+    let iter = config.db.iterator(IteratorMode::Start);
+    let mut item_count = 0;
+    let mut remove_count = 0;
+    for (id, value) in iter {
+        item_count += 1;
+        let mut content_id = [0u8; 32];
+        content_id.copy_from_slice(&id);
+        if purge_config.mode == "purge" {
+            match storage.evict(content_id) {
+                Ok(_) => remove_count += 1,
+                Err(err) => println!(
+                    "Error occurred while evicting content id: {:?} - {:?}",
+                    content_id, err
+                ),
+            }
+        } else {
+            // naked unwrap since we shouldn't be storing any invalid content keys
+            let key = storage.lookup_content_key(content_id).unwrap();
+            let content_key = HistoryContentKey::try_from(key).unwrap();
+            if !is_content_valid(&content_key, &value.into_vec()) {
+                match storage.evict(content_id) {
+                    Ok(_) => remove_count += 1,
+                    Err(err) => println!(
+                        "Error occurred while evicting content key: {:?} - {:?}",
+                        content_key, err
+                    ),
+                }
+            }
+        }
+    }
+    println!(
+        "Found {:?} total items - Removed {:?} items",
+        item_count, remove_count
+    );
+    Ok(())
+}
+
+fn is_content_valid(content_key: &HistoryContentKey, value: &[u8]) -> bool {
+    match content_key {
+        HistoryContentKey::BlockHeaderWithProof(_) => {
+            if BlockHeaderWithProof::from_ssz_bytes(value).is_err() {
+                return false;
+            }
```

Comment:
we could simplify all match arms and remove the `true` at the bottom

```suggestion
            BlockHeaderWithProof::from_ssz_bytes(value).is_ok()
```